library/src/main/scala/scala/scalajs/js/annotation/LinkTime.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.scalajs.js.annotation
+
+/** Specifies that the entity can be replaced by a value known at link-time.
+ *
+ *  The annotated entity can be used in the condition of `LinkingInfo.linkTimeIf`.
+ *  When linking, the annotated entity will be replaced by a value determined
+ *  at link-time, as specified in `LinkTimeProperties.scala`.
+ *
+ *  @see [[LinkingInfo.linkTimeIf]]
+ *  @see [[LinkTimeProperties]]
+ */
+private[scalajs] class LinkTime extends scala.annotation.StaticAnnotation
```

Comment:
I know the `@JSxyz` annotations belie me, but the general convention in Scala is for annotations to start with lowercase. So consider calling this `linkTime` instead.

Maybe this should even be called `linkTimeProperty`?